src/Integration/MefServices/DeprecatedSonarRuleSetManager.cs

```diff
@@ -1,186 +0,0 @@
-            this.activeSolutionTracker.ActiveSolutionChanged += OnActiveSolutionChanged;
-
-            if (this.activeSolutionBoundTracker.CurrentConfiguration != null &&
-                this.activeSolutionBoundTracker.CurrentConfiguration.Mode != NewConnectedMode.SonarLintMode.LegacyConnected)
-            {
-                WarnIfAnyProjectHasSonarRuleSet();
-            }
-        }
-
-        private void OnActiveSolutionChanged(object sender, ActiveSolutionChangedEventArgs e)
-        {
-            try
-            {
-                if (e.IsSolutionOpen)
-                {
-                    WarnIfAnyProjectHasSonarRuleSet();
-                }
-            }
-            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
-            {
-                // Swallow the exception as we are on UI thread
-            }
-        }
-
-        private void OnSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
-        {
-            try
-            {
-                if (e.Configuration != null &&
-                    e.Configuration.Mode != NewConnectedMode.SonarLintMode.LegacyConnected)
-                {
-                    WarnIfAnyProjectHasSonarRuleSet();
-                }
-            }
-            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
-            {
-                // Swallow the exception as we are on UI thread
-            }
-        }
-
-        private void WarnIfAnyProjectHasSonarRuleSet()
-        {
-            var hasAnySonarRule = this.projectSystemHelper.GetSolutionProjects()
-                .SelectMany(p => ruleSetProvider.GetProjectRuleSetsDeclarations(p))
-                .Any(HasAnySonarRule);
-
-            if (hasAnySonarRule)
-            {
-                this.logger.WriteLine(DeprecationMessage);
-            }
-        }
-
-        private bool HasAnySonarRule(RuleSetDeclaration ruleSetDeclaration)
-        {
-            var projectDirectoryFullPath = Path.GetDirectoryName(ruleSetDeclaration.RuleSetProjectFullName);
-            var projectRuleSetFullPath = GetFullPath(ruleSetDeclaration.RuleSetPath, projectDirectoryFullPath);
-            if (!File.Exists(projectRuleSetFullPath))
-            {
-                return false;
-            }
-
-            var projectRuleSet = RuleSet.LoadFromFile(projectRuleSetFullPath);
-
-            // 1. Collect all paths (current ruleset + includes)
-            var ruleSetIncludeFullPaths = projectRuleSet.RuleSetIncludes
-                .Select(include => GetFullPath(include.FilePath, projectDirectoryFullPath))
-                .ToList();
-
-            // 2. Look if any of the effective rules is from SonarAnalyzer and initialize dictionary with this result.
-            return projectRuleSet
-                .GetEffectiveRules(ruleSetIncludeFullPaths, new RuleInfoProvider[0])
-                .Any(rule => rule.AnalyzerId.StartsWith("SonarAnalyzer.", StringComparison.OrdinalIgnoreCase));
```

Comment:
FYI this line was throwing if the ruleset contained an included ruleset that could not be located...